programs/github

```diff
@@ -1,8 +1,38 @@
 local github = dofile("apis/github")
 
-local action = select(1, ...)
+local selectedAction = select(1, ...)
+local passed = { select(2, ...) }
 
-local usage = "Usage: github clone <user>/<repo> [<branch>|<tag>] [<destination>]"
+local subcommands = {
+	clone = {
+		name = 'Clone',
+		usage = "github clone <user>/<repo> [-b <branchname> | -t <tagname>] [<destination>] [-a <username>]",
+		flags = {
+			['-b'] = 'branch',
+			['--branch'] = 'branch',
+			['-t'] = 'tag',
+			['--tag'] = 'tag',
+			['-a'] = 'authed',
+			['-auth'] = 'authed'
+		}
+	},
+	auth = {
+		name = 'Auth',
+		usage = "github auth <user> [<api token> | -d ]",
+		flags = {
+			['-d'] = 'delete',
+			['--delete'] = 'delete'
+		}
+	}
+}
+
+local action = subcommands[selectedAction]
+if not action then
+	for k, v in pairs(subcommands) do
+		print(v.usage)
```

Comment:
Perhaps manually wrapping the usage strings neatly is the right thing to do then? Unless CC screens can be different sizes. Either way, that's a job for another PR.